llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -4812,6 +4812,8 @@ bool RISCV::isVLKnownLE(const MachineOperand &LHS, const MachineOperand &RHS) {
     return true;
   if (RHS.isImm() && RHS.getImm() == RISCV::VLMaxSentinel)
     return true;
+  if (LHS.isImm() && LHS.getImm() == 0)
+    return true;
   if (LHS.isImm() && LHS.getImm() == RISCV::VLMaxSentinel)
```

Comment:
Not related to this patch, but LHS being RISCV::VLMaxSentinel doesn't rule out RHS also being dynamically VLMax does it? So they could still be equal. Is it correct to return false here?